public/app/plugins/panel/alertlist/unified-alerting/UngroupedView.tsx

```diff
@@ -27,86 +31,176 @@ type UngroupedModeProps = {
   options: UnifiedAlertListOptions;
 };
 
+export const useGetExtraInfoForAlertItem = (returnTo: string, slicedRules: CombinedRuleWithLocation[]) => {
+  const [rulesToDisplay, setRulesToDisplay] = useState<CombinedRuleWithLocationExtended[] | undefined>(undefined);
+  useEffect(() => {
+    const getExtraInfo = async () => {
+      const rulesWithExtraInfo: CombinedRuleWithLocationExtended[] = await Promise.all(
+        slicedRules.map(async (rule: CombinedRuleWithLocation) => {
+          const identifier = fromCombinedRule(rule.dataSourceName, rule);
+          const strIndentifier = stringifyIdentifier(identifier);
+
+          const refToRuleView = createUrl(
+            `/alerting/${encodeURIComponent(rule.dataSourceName)}/${encodeURIComponent(strIndentifier)}/view`,
+            { returnTo: returnTo ?? '' }
+          );
+          type HistoryResponse = {
+            data: { values: string[][] };
+          };
+          let history: HistoryResponse | undefined = undefined;
+          // history is supported only for grafana-managed alerts so far.
+          if (isGrafanaRuleIdentifier(identifier)) {
+            const now = new Date().getTime();
+            const aWeekAgo = addDurationToDate(now, { days: -7 }).getTime();
+            const timeRange: RelativeTimeRange = {
+              from: aWeekAgo,
+              to: now,
+            };
+            history = await getBackendSrv().get('/api/v1/rules/history', { ruleUID: strIndentifier, ...timeRange });
+          }
+          let firingCount =
+            history &&
+            history.data?.values[2].reduce(
+              (sum, state, index) => (state === GrafanaAlertState.Alerting ? sum + 1 : sum),
+              0
+            );
+
+          const ruleToDisplay: CombinedRuleWithLocationExtended = {
+            ...rule,
+            refToRuleView: refToRuleView,
+            firingCount: firingCount,
+          };
+          return ruleToDisplay;
+        })
+      );
+      return rulesWithExtraInfo;
+    };
+
+    getExtraInfo().then((rules) => setRulesToDisplay(rules));
```

Comment:
Before you use `await`, can we do the same here instead of `.then` for consistency?